src/test/ui/closures/issue-84366-closure-outlives-ret.rs

```diff
@@ -0,0 +1,37 @@
+// Regression test for issue #84366
+// Ensures that proving `closure: 'a` requires that
+// we prove `closure_ret_type: 'a`
+
+use std::fmt;
```

Comment:
This test should stop compiling with your  PR as is

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=072f28a4b187fb5f912706353ba1ac41

But I believe this test will still compile until you fix the yield case

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=5cc9c10b51c12546a8457676b68e296d